numpy/core/src/umath/ufunc_object.c

```diff
@@ -5017,6 +5200,7 @@ ufunc_dealloc(PyUFuncObject *ufunc)
 {
     PyArray_free(ufunc->core_num_dims);
     PyArray_free(ufunc->core_dim_ixs);
+    PyArray_free(ufunc->core_dim_szs);
```

Comment:
Should also free `->core_dim_flexible` or we'll have a memory leak...

Reply:
fixed